plugin/src/main/java/org/opennms/plugins/cloud/tsaas/GrpcConnection.java

```diff
@@ -0,0 +1,165 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.plugins.cloud.tsaas;
+
+import static org.opennms.plugins.cloud.tsaas.SecureCredentialsVaultUtil.SCV_ALIAS;
+import static org.opennms.plugins.cloud.tsaas.SecureCredentialsVaultUtil.Type.privatekey;
+import static org.opennms.plugins.cloud.tsaas.SecureCredentialsVaultUtil.Type.publickey;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLException;
+
+import org.opennms.integration.api.v1.scv.Credentials;
+import org.opennms.plugins.cloud.tsaas.grpc.comp.ZStdCodecRegisterUtil;
+import org.opennms.tsaas.TimeseriesGrpc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.ClientAuth;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslContextBuilder;
+import io.grpc.netty.shaded.io.netty.handler.ssl.SslProvider;
+
+public class GrpcConnection {
+    private static final Logger LOG = LoggerFactory.getLogger(GrpcConnection.class);
+    // 100M sync with cortex server
+    private final static int MAX_MESSAGE_SIZE = 104857600;
+
+    private ManagedChannel managedChannel;
+    private TimeseriesGrpc.TimeseriesBlockingStub clientStub;
+
+    public void init(final TsaasConfig config, final SecureCredentialsVaultUtil scvUtil) {
```

Comment:
Why not make it as a constructor? So that no one will miss calling init. Also you don't need to do the destroy in init.

Reply:
Also I think the params better to do null check.